### ISSUE TITLE
Scripts and functions got `real` again

### ISSUE THREAD

**Issue by thennothinghappened** [YoYoGames/GameMaker-Bugs#5724]
### Description
Script assets and functions are no longer handles and have become numbers again, see attached project or below GML.

### Steps To Reproduce
**`Script1.gml`**:
```gml
function blah() {}

show_message(is_handle(is_handle));
show_message(is_handle(Script1));
show_message(is_handle(blah));

// 2024.2:
// 0
// 1
// 1

// 2024.4:
// 0
// 0
// 0
```

### Which version of GameMaker are you reporting this issue for?
IDE v2024.400.0.556 Runtime v2024.400.0.571

### Which operating system(s) are you seeing the problem on?
Mac 13.5

### Which platform(s) are you seeing the problem on?
HTML5, macOS

[69e83c09-6370-4aff-8487-ca4b233784c9](https://api.gamemaker.io/api/github/downloads/69e83c09-6370-4aff-8487-ca4b233784c9)

**Comment by rwkay** [comment]
Fixed in 2024.6

**Comment by backYard321** [comment]
It's not just `is_handle()` - `show_message(scriptname)` will show a number and not a ref.

**Comment by alicemoretti** [comment]
Verified fixed on IDE v2024.4.1.141  Runtime v2024.4.1.174.

Thank you.
